plugins/modules/ec2_asg_tag.py

```diff
@@ -0,0 +1,237 @@
+description:
+  - Creates, modifies and removes tags for AutoScaling Groups
+author: "Jonathan Sokolowski (@jsok)"
+requirements: [ "boto3", "botocore" ]
+options:
+  name:
+    description:
+      - The ASG name.
+    required: true
+    type: str
+  state:
+    description:
+      - Whether the tags should be present or absent on the ASG.
+    default: present
+    choices: ['present', 'absent']
+    type: str
+  tags:
+    description:
+      - A list of tags to add or remove from the ASG.
+      - If the value provided for a key is not set and I(state=absent), the tag will be removed regardless of its current value.
+      - Optional key is I(propagate_at_launch), which defaults to true.
+    type: list
+    elements: dict
+  purge_tags:
+    description:
+      - Whether unspecified tags should be removed from the resource.
+      - Note that when combined with I(state=absent), specified tags with non-matching values are not purged.
+    type: bool
+    default: false
+extends_documentation_fragment:
+- amazon.aws.aws
+- amazon.aws.ec2
+'''
+
+EXAMPLES = r'''
+- name: Ensure tags are present on an ASG
+  community.aws.ec2_asg_tag:
+    name: my-auto-scaling-group
+    state: present
+    tags:
+      - environment: production
+        propagate_at_launch: true
+      - role: webserver
+        propagate_at_launch: true
+
+- name: Ensure tag is absent on an ASG
+  community.aws.ec2_asg_tag:
+    name: my-auto-scaling-group
+    state: absent
+    tags:
+      - environment: development
+
+- name: Remove all tags except Name from an ASG
+  community.aws.ec2_asg_tag:
+    name: my-auto-scaling-group
+    state: absent
+    tags:
+      - Name: ''
+    purge_tags: true
+'''
+
+RETURN = r'''
+---
+tags:
+  description: A list containing the tags on the resource
+  returned: always
+  type: list
+  sample: [
+      {
+          "key": "Name",
+          "value": "public-webapp-production-1",
+          "resource_id": "public-webapp-production-1",
+          "resource_type": "auto-scaling-group",
+          "propagate_at_launch": "true"
+      },
+      {
+          "key": "env",
+          "value": "production",
+          "resource_id": "public-webapp-production-1",
+          "resource_type": "auto-scaling-group",
+          "propagate_at_launch": "true"
+      }
+  ]
+added_tags:
+  description: A list of tags that were added to the ASG
+  returned: If tags were added
+  type: list
+removed_tags:
+  description: A list of tags that were removed from the ASG
+  returned: If tags were removed
+  type: list
+'''
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+from ansible.module_utils._text import to_native
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+
+
+def to_boto3_tag_list(tags, group_name):
```

Comment:
Could you please import and use the `ansible_dict_to_boto3_tag_list` function from the module utility?